flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlNodeToOperationConversion.java

```diff
@@ -1379,6 +1379,6 @@ private String getQuotedSqlString(SqlNode sqlNode) {
     private PlannerQueryOperation toQueryOperation(FlinkPlannerImpl planner, SqlNode validated) {
         // transform to a relational tree
         RelRoot relational = planner.rel(validated);
-        return new PlannerQueryOperation(relational.project());
+        return new PlannerQueryOperation(relational.project(), getQuotedSqlString(validated));
```

Comment:
`SqlNodeToOperationConversion#toQueryOperation` seems only used in explain. I'm not sure how you did the manually testing, IIUC, the job name cannot be changed for select queries?